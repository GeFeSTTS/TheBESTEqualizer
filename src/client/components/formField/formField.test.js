import checkPropTypes from 'check-prop-types';
import React from 'react';
import { shallow } from 'enzyme';
import FormField from './formField';

const defaultProps = {
  onInputChange: jest.fn(),
  value: '',
  error: '',
  field: {
    name: '',
    label: '',
    type: '',
  },
};

const checkProps = (component, expectedProps) => {
  const propError = checkPropTypes(component.propTypes, expectedProps, 'prop', component.name);
  expect(propError).toBeUndefined();
};

describe('testing render', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<FormField {...defaultProps} />);
  });
  it('should render formField properly', () => {
    expect(wrapper.debug()).toMatchSnapshot();
  });
  it('should call "onInputChange" function', () => {
    wrapper.find('input').simulate('change');
    expect(defaultProps.onInputChange).toHaveBeenCalled();
  });
});

describe('testing ptopTypes', () => {
  it('does not throw warning with expected props', () => {
    checkProps(FormField, defaultProps);
  });
});
