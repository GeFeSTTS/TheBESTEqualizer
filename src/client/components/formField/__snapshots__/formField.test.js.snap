// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing render should render formField properly 1`] = `
"<div className=\\"formField\\">
  <label htmlFor=\\"field\\" className=\\"label\\" />
  <div className=\\"field\\">
    <input onChange={[Function: mockConstructor]} name=\\"\\" type=\\"\\" value=\\"\\" />
  </div>
  <div className=\\"error\\" />
</div>"
`;
